src/Symfony/Component/ObjectMapper/Metadata/Mapping.php

```diff
@@ -23,7 +23,7 @@
     /**
      * @param string|class-string|null                                                               $source    The property or the class to map from
      * @param string|class-string|null                                                               $target    The property or the class to map to
-     * @param string|bool|callable(mixed, object): bool|null                                         $if        A boolean, Symfony service name or a callable that instructs whether to map
+     * @param class-string|string|bool|callable(mixed, object): bool|null                            $if        A boolean, Symfony service name or a callable that instructs whether to map
      * @param (string|callable(mixed, object): mixed)|(string|callable(mixed, object): mixed)[]|null $transform A service id or a callable that transform the value during mapping
      */
     public function __construct(
```

Comment:
did you consider using class Attribute\Map instead of this Mapping class? Looks like they hold the exact same data.
Alternatively, have class Mapping extend Map to skip the duplicate declaration (this is all internal so the type hierarchy doesn't matter).

Reply:
Yes I think this was asked for in the original PR I'd have used the same class for both metadata and attributes I don't see the need for 2 classes...

Reply:
@chalasr could I have your opinion on this? I think we could merge the `Mapping` and the `Map` but I think you had arguments not to do so.

Reply:
I'll open a new pr for that.